plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -2732,26 +2732,29 @@ public void testAllAvailableTypes()
         assertUpdate("DROP TABLE test_all_types");
     }
 
-    @Test
+    @Test(invocationCount = 10)
     public void testLocalDynamicFilteringWithSelectiveBuildSizeJoin()
     {
         long fullTableScan = (Long) computeActual("SELECT count(*) FROM lineitem").getOnlyValue();
-        long numberOfFiles = (Long) computeActual("SELECT count(*) FROM \"lineitem$files\"").getOnlyValue();
+        // Pick a value for totalprice where file level stats will not be able to filter out any data
+        MaterializedRow range = getOnlyElement(computeActual("SELECT max(lower_bounds[4]), min(upper_bounds[4]) FROM \"orders$files\"").getMaterializedRows());
+        double totalPrice = (Double) computeActual(format(
+                "SELECT totalprice FROM orders WHERE totalprice > %s AND totalprice < %s LIMIT 1",
+                range.getField(0),
+                range.getField(1)))
+                .getOnlyValue();
```

Comment:
But why are we sure gonna find it?
If we have two files with non overlaping ranges we get empty result range.

E.g: 
* f1: [10, 20]
* f2: [3,7]

We get range from the first query: [10, 7]. Which yields empty result from second query.

I am probably missing sth.

Reply:
That is possible, and would break the test, but seems unlikely. The overlapping range when I ran the test was most of the range of the column. Unless the data is explicitly ordered I think there should be some overlap.

If this still feels too flaky we can take it out and try testing the DF a different way.